### PR TITLE
[office-js][office-js-preview] (common) Document device permission API

### PR DESCRIPTION
**Pull request by samantharamon**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: None.
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by samantharamon** [comment]
@AlexJerabek tagging for review.

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -509,6 +509,32 @@ declare namespace Office {
          */
         Text,
     }
+    /**
+     * Specifies the device capability to which an add-in is requesting access.
+     *
+     * @remarks
+     *
+     * **Applications**: Excel on the web, Outlook on the web, PowerPoint on the web, and Word on the web running in Chromium-based browsers,
+     * such as Microsoft Edge and Google Chrome.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:518]: Please make the same browser-first change here as elsewhere.

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -509,6 +509,32 @@ declare namespace Office {
...
+     */
+    enum DevicePermissionType {
+        /**
+         * The add-in is requesting access to the user's camera.
+         */
+        camera,
+        /**
+         * The add-in is requesting access to the user's geolocation.
+         *
+         * **Important**: Access to a user's geolocation is only supported in add-ins running in Outlook on the web.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:530]: I think the add-in's context here is clear.
```suggestion
         * **Important**: Access to a user's geolocation is only supported in Outlook on the web.
```

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -5204,6 +5230,156 @@ declare namespace Office {
...
+     * **Requirement set**: {@link https://learn.microsoft.com/javascript/api/requirement-sets/common/device-permission-service-requirement-sets | DevicePermission 1.1}
+     */
+    interface DevicePermission {
+        /**
+         * Requests permission from a user to access their device capabilities, such as a camera or microphone.
+         *
+         * All the requested permissions are displayed in a single modal dialog to the user.
+         * The dialog includes options to **Allow**, **Allow once**, or **Deny** the requested permissions.
+         *
+         * This method returns a promise and is supported in web versions of Excel, PowerPoint, and Word running in a Chromium-based browser, such as Microsoft Edge or Google Chrome.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:5259]: I think you can let the interface description do the heavy-lifting and avoid repeating all the details here. That way, the focus can be on this being for WXP.
```suggestion
         * This method returns a promise. Use it with Excel, PowerPoint, and Word add-ins.
```

File: types/office-js-preview/index.d.ts [hunk 4]
```diff
@@ -5204,6 +5230,156 @@ declare namespace Office {
...
+         *
+         * **Applications**: This API is supported by the following Office applications when running in Chromium-based browsers, such as Microsoft Edge and Google Chrome.
+         *
+         * - Excel on the web
+         *
+         * - PowerPoint on the web
+         *
+         * - Word on the web
+         *
+         * **Requirement set**: {@link https://learn.microsoft.com/javascript/api/requirement-sets/common/device-permission-service-requirement-sets | DevicePermission 1.1}
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:5271]: I think this is fine to omit at the per-method level.
```suggestion

```

File: types/office-js-preview/index.d.ts [hunk 5]
```diff
@@ -5204,6 +5230,156 @@ declare namespace Office {
...
+         *     Access to a user's geolocation is blocked.
+         */
+        requestPermissions(permissions: Office.DevicePermissionType[]): Promise<boolean>;
+        /**
+         * Requests permission from a user to access their device capabilities, such as a camera, geolocation, or microphone.
+         *
+         * All the requested permissions are displayed in a single modal dialog to the user.
+         * The dialog includes options to **Allow**, **Allow once**, or **Deny** the requested permissions.
+         *
+         * This method accepts a callback function and is supported in Outlook on the web running in a Chromium-based browser, such as Microsoft Edge or Google Chrome.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:5303]:
```suggestion
         * This method accepts a callback function. Use it with Outlook add-ins.
```

File: types/office-js-preview/index.d.ts [hunk 6]
```diff
@@ -5204,6 +5230,156 @@ declare namespace Office {
...
+         *
+         * @remarks
+         *
+         * **Applications**: Outlook on the web running in Chromium-based browsers, such as Microsoft Edge and Google Chrome.
+         *
+         * **Requirement set**: {@link https://learn.microsoft.com/javascript/api/requirement-sets/common/device-permission-service-requirement-sets | DevicePermission 1.1}
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:5309]:
```suggestion

```

File: types/office-js-preview/index.d.ts [hunk 7]
```diff
@@ -5204,6 +5230,156 @@ declare namespace Office {
...
+         *     If the user grants permission to access the requested device capabilities, `true` is returned in the `asyncResult.value` property.
+         */
+        requestPermissionsAsync(permissions: Office.DevicePermissionType[], options: Office.AsyncContextOptions, callback: (asyncResult: Office.AsyncResult<boolean>) => void): void;
+        /**
+         * Requests permission from a user to access their device capabilities, such as a camera, geolocation, or microphone.
+         *
+         * All the requested permissions are displayed in a single modal dialog to the user.
+         * The dialog includes options to **Allow**, **Allow once**, or **Deny** the requested permissions.
+         *
+         * This method accepts a callback function and is supported in Outlook on the web running in a Chromium-based browser, such as Microsoft Edge or Google Chrome.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:5346]:
```suggestion
         * This method accepts a callback function. Use it with Outlook add-ins.
```

File: types/office-js-preview/index.d.ts [hunk 8]
```diff
@@ -5204,6 +5230,156 @@ declare namespace Office {
...
+         *
+         * @remarks
+         *
+         * **Applications**: Outlook on the web running in Chromium-based browsers, such as Microsoft Edge and Google Chrome.
+         *
+         * **Requirement set**: {@link https://learn.microsoft.com/javascript/api/requirement-sets/common/device-permission-service-requirement-sets | DevicePermission 1.1}
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:5352]:
```suggestion

```

**Review by AlexJerabek — CHANGES_REQUESTED** [review]
Some suggestions. Please repeat them in the other d.ts file too.

File: types/office-js-preview/index.d.ts (hunk 1 above)
Comment (samantharamon) [reply types/office-js-preview/index.d.ts:518]: Ah, thanks for catching this. Updated.

**Review by AlexJerabek — APPROVED** [review]

**Review by Rick-Kirkham — APPROVED** [review]

**Pull request merged**